analyses/org.eclipse.tracecompass.incubator.dpdk.core/src/org/eclipse/tracecompass/incubator/internal/dpdk/core/lcore/analysis/DpdkLogicalCoreEventHandler.java

```diff
@@ -0,0 +1,58 @@
+/**********************************************************************
+ * Copyright (c) 2024 École Polytechnique de Montréal
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+package org.eclipse.tracecompass.incubator.internal.dpdk.core.lcore.analysis;
+
+import java.util.Objects;
+
+import org.eclipse.tracecompass.incubator.internal.dpdk.core.analysis.IDpdkEventHandler;
+import org.eclipse.tracecompass.incubator.internal.dpdk.core.lcore.analysis.LogicalCore.LogicalCoreRole;
+import org.eclipse.tracecompass.incubator.internal.dpdk.core.lcore.analysis.LogicalCore.LogicalCoreStatus;
+import org.eclipse.tracecompass.statesystem.core.ITmfStateSystemBuilder;
+import org.eclipse.tracecompass.tmf.core.event.ITmfEvent;
+
+/**
+ * Event handler to handle core related events
+ *
+ * @author Adel Belkhiri
+ * @author Arnaud Fiorini
+ */
+public class DpdkLogicalCoreEventHandler implements IDpdkEventHandler {
+
+    private DpdkLogicalCoreEventLayout fLayout;
+
+    DpdkLogicalCoreEventHandler(DpdkLogicalCoreEventLayout layout) {
+        fLayout = layout;
+    }
+
+    @Override
+    public void handleEvent(ITmfStateSystemBuilder ssb, ITmfEvent event) {
+        Integer lcoreId = event.getContent().getFieldValue(Integer.class, fLayout.fieldLcoreId());
+        long ts = event.getTimestamp().getValue();
+        String eventName = event.getName();
+
+        if (eventName.equals(fLayout.eventLcoreStateChange())) {
```

Comment:
How about fLayout.isLcoreStateChange(event)?

Reply:
I would argue that it is better to have everything there since the logic is pretty simple. If the logic would be more complex, moving those elsewhere might be interesting to encapsulate the logic, but here it would actually make it harder to see the logic behind which events is causing a state change or indicates that the lcore is running etc...